tests/Doctrine/Tests/ORM/Functional/SQLFilterWithFetchEAGERTest.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\Query\Filter\SQLFilter;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class SQLFilterWithFetchEAGERTest extends OrmFunctionalTestCase
+{
+    /**
+     * @var int
+     */
+    private $company1Id;
+
+    /**
+     * @var int
+     */
+    private $patient1Id;
+
+    /**
+     * @var int
+     */
+    private $insurance1Id;
+
+    /**
+     * @var int
+     */
+    private $company2Id;
+
+    /**
+     * @var int
+     */
+    private $patient2Id;
+
+    /**
+     * @var int
+     */
+    private $insurance2Id;
+
+    private function createModels(): void
+    {
+        $company      = new Company();
+        $patient      = new Patient($company);
+        $insurance    = new Insurance($company, 'name1');
+        $patInsurance = new PatientInsurance($patient, $insurance, 'test1');
+
+        $company2      = new Company();
+        $patient2      = new Patient($company2);
+        $insurance2    = new Insurance($company2, 'name2');
+        $patInsurance2 = new PatientInsurance($patient2, $insurance2, 'test1');
+
+        $this->_em->persist($company);
+        $this->_em->persist($patient);
+        $this->_em->persist($patInsurance);
+        $this->_em->persist($insurance);
+
+        $this->_em->persist($company2);
+        $this->_em->persist($patient2);
+        $this->_em->persist($patInsurance2);
+        $this->_em->persist($insurance2);
+
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $this->company1Id   = $company->id;
+        $this->patient1Id   = $patient->id;
+        $this->insurance1Id = $insurance->id;
+
+        $this->company2Id   = $company2->id;
+        $this->patient2Id   = $patient2->id;
+        $this->insurance2Id = $insurance2->id;
```

Comment:
Is ID generation actually part of the issue? If not, use fixed IDs instead of creating all these properties on the test case.

Reply:
fixed